Add qk norm optionally before attention calculation

madhu-fb opened the pull request:
Differential Revision: D70355802

madhu-fb commented:
@pytorchbot label "topic: not user facing"

iseeyuan reviewed (approved):
LGTM. Thank you @madhu-fb !

madhu-fb committed:
Add qk norm optionally before attention calculation (#8820)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/8820

Some of the new llama checkpoints developed by genai use an additional qk_norm in the attention calculation. To run these models with executorch and have parity with server models, we require an optional qk norm in the ET attention.

Refactoring RMSNorm into a separate file so that there is no circular dependency between attention and llama_transformer

Reviewed By: iseeyuan

Differential Revision: D70355802
